crates/claims/crates/vc/src/v1/syntax/credential.rs

```diff
@@ -101,43 +115,68 @@ pub struct SpecializedJsonCredential<S = json_syntax::Object, C = (), T = ()> {
         default,
         skip_serializing_if = "Vec::is_empty"
     )]
-    pub terms_of_use: Vec<MaybeIdentifiedTypedObject>,
+    pub terms_of_use: Vec<TermsOfUse>,
 
     /// Evidence.
     #[serde(
         with = "value_or_array",
         default,
         skip_serializing_if = "Vec::is_empty"
     )]
-    pub evidence: Vec<MaybeIdentifiedTypedObject>,
+    pub evidence: Vec<Evidence>,
 
     #[serde(rename = "credentialSchema")]
     #[serde(
         with = "value_or_array",
         default,
         skip_serializing_if = "Vec::is_empty"
     )]
-    pub credential_schema: Vec<IdentifiedTypedObject>,
+    pub credential_schema: Vec<Schema>,
 
     #[serde(rename = "refreshService")]
     #[serde(
         with = "value_or_array",
         default,
         skip_serializing_if = "Vec::is_empty"
     )]
-    pub refresh_services: Vec<IdentifiedTypedObject>,
+    pub refresh_services: Vec<RefreshService>,
 
     #[serde(flatten)]
     pub additional_properties: BTreeMap<String, json_syntax::Value>,
```

Comment:
Might as well genericise this as well

Reply:
You mean the `json_syntax::Value` part or the whole map?

Reply:
I meant the whole map. But it makes it possible then for users to make mistakes, unless `flatten` would catch it if it's not an object-like type -- I just think it would be more consistent, but I don't feel that strongly about it

Reply:
`flatten` will catch it at runtime if the type is not a map or unit. I made the change, I think this is a good idea that allows two use cases:
- Create a credential type that disallow extra properties, by setting the `ExtraProperties` type parameter to an empty struct desirializing with the attribute `deny_unknown_fields`.
- Create a credential type with required extra properties by setting it to a custom struct.